tests/suites/test_suite_psa_crypto_metadata.data

```diff
@@ -26,6 +26,22 @@ Hash: SHA-2 SHA-512
 depends_on:PSA_WANT_ALG_SHA_512
 hash_algorithm:PSA_ALG_SHA_512:64
 
+Hash: SHA-3 SHA3-224
+depends_on:PSA_WANT_ALG_SHA_3
+hash_algorithm:PSA_ALG_SHA3_224:28
+
+Hash: SHA-3 SHA3-256
+depends_on:PSA_WANT_ALG_SHA_3
+hash_algorithm:PSA_ALG_SHA3_256:32
+
+Hash: SHA-3 SHA3-384
+depends_on:PSA_WANT_ALG_SHA_3
+hash_algorithm:PSA_ALG_SHA3_384:48
+
+Hash: SHA-3 SHA3-512
+depends_on:PSA_WANT_ALG_SHA_3
+hash_algorithm:PSA_ALG_SHA3_512:64
+
```

Comment:
Should we also add tests for sha3 tests for `hmac_algorithm` and other supported algorithms?

Reply:
There doesn't seem to be a lot of existing test coverage of PSA HMAC - there's no where obvious to add cases for HMAC SHA3 (and in any case, this probably isn't a commonly used combination as [HMAC isn't useful for SHA-3](https://en.wikipedia.org/wiki/HMAC#:~:text=The%20Keccak%20hash%20function%2C%20that%20was%20selected%20by%20NIST%20as%20the%20SHA%2D3%20competition%20winner%2C%20doesn%27t%20need%20this%20nested%20approach%20and%20can%20be%20used%20to%20generate%20a%20MAC%20by%20simply%20prepending%20the%20key%20to%20the%20message%2C%20as%20it%20is%20not%20susceptible%20to%20length%2Dextension%20attacks.%5B9%5D).

Reply:
I've added HMAC test cases, along with cases for the other hash algorithms.

Reply:
> HMAC isn't useful for SHA-3

HMAC isn't useful for SHA-3 because the simple construction SHA3(K||M) is enough to construct a secure MAC (with a fixed-size key K), and NIST has standardized KMAC which is essentially that (with just a few extra bits getting hashed to provide domain separation).

However, in practice, HMAC is used with SHA3, because there's a lot of code out there that builds HMAC on top of a hash black box that can be instantiated either with SHA3 or with other hashes for which H(K||M) is not secure.

The PSA tests don't try to test underlying algorithms extensively because it relies on tests of lower-level modules for that. However, since the PSA code has its own implementation of HMAC, the HMAC tests in `test_suite_md` don't help, and so the PSA tests should have tests of HMAC with all underlying hashes. It currently doesn't, that's a gap in our test coverage, thanks for resolving it.

Reply:
Thanks for the explanation and the test cases :)